modules/features/lfc_checker/src/lfc_checker.cpp

```diff
@@ -0,0 +1,232 @@
+#include "lfc_checker.h"
+#include "controller.h"
+#include "fifo_queue.h"
+#include <cstdio>
+#include "libtp_c/include/d/com/d_com_inf_game.h"
+#include "libtp_c/include/f_op/f_op_actor_mng.h"
+#include "libtp_c/include/m_Do/m_Do_printf.h"
+#include "libtp_c/include/d/d_procname.h"
+#include "libtp_c/include/SSystem/SComponent/c_counter.h"
+
+enum Phase {
+    LADDER_START,
+    LADDER_DURING,
+    LADDER_DURING_2,
+    LADDER_DROP,
+    LADDER_DROP_2,
+    LADDER_REGRAB
+};
+
+u8 phase = Phase::LADDER_START;
+int frameDelta = 0;
+s32 drop_1_start_frame;       // The frame the first drop started on.
+s32 drop_2_start_frame;         // The frame the second drop ended on.
+bool game_paused;
+
+fopAc_ac_c* findLadderFreezard() {
```

Comment:
i think this function could probably be simplified into a search callback rather than reimplementing the entire search process. you'd basically do something like
```c++
fopAc_ac_c* freezard = (fopAc_ac_c*)fpcM_Search(searchLadderFreezard, nullptr);
```
with the the callback like
```c++
void* searchLadderFreezard(void* i_actor, void* i_data) {
    // ...
}
```
where i_actor is the iterating actor

Reply:
yea i discovered this after i implemented my logic lol. will update it to this